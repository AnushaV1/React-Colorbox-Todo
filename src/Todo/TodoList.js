import React, {useState} from "react";
import Todo from "./Todo";
import TodoForm from './TodoForm'
const TodoList = () => {

    const [todos, setTodos] = useState([]);
    
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, newTodo]);
    }

    const remove = id => {
        setTodos(todos => todos.filter(todo => todo.id !==id))
    }

    return (
    <>
        <TodoForm addTodo={addTodo}/>
    <div>    <ul>
    {todos.map(todo => (
    <Todo
        remove={remove}
        key={todo.id}
        id={todo.id}
        task={todo.task}
    />
    ))
    }
    </ul>
    </div>
    </>

    )
}

export default TodoList;