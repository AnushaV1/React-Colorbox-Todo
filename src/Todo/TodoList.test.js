import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from './TodoList';


function addTodo(todoList, task = "write tests") {
    const taskInput = todoList.getByLabelText("Add Task");
    fireEvent.change(taskInput, {target: { value: task }});
    const submitBtn = todoList.getByText("Add To List");
    fireEvent.click(submitBtn);
}


it("renders without crashing", function() {
    render(<TodoList />);
});

it("matches snapshot", function() {
    const {asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("add a todo to list", function() {
    const list = render(<TodoList />);
    addTodo(list);

    expect(list.getByLabelText('Add Task')).toHaveValue('');
    expect(list.getByText('write tests')).toBeInTheDocument();
    expect(list.getByText('X')).toBeInTheDocument();
});

it('delete on click a todo', function() {
    const list = render(<TodoList />);
    addTodo(list);
    fireEvent.click(list.getByText('X'));
    expect(list.queryByText("write tests")).not.toBeInTheDocument();
})