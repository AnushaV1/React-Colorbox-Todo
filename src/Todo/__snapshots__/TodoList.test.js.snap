// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Add Task
    </label>
    <input
      id="task"
      name="task"
      placeholder="Enter Todo item"
      type="text"
      value=""
    />
    <button>
      Add To List
    </button>
  </form>
  <div>
        
    <ul />
  </div>
</DocumentFragment>
`;
